### PR TITLE
Update Ballerina By Examples to include references to the `binary data` BBE links

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Update Ballerina By Examples to include references to the binary data and list equality BBE links

Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/9385

### CODE REVIEW

**Review by sm1990 — APPROVED** [review]

File: examples/list-equality/list_equality.md [hunk 1]
```diff
@@ -10,5 +10,3 @@ You can use `==` and `!=` on lists to check the deep equality of two lists: two
 - [Tuples](/learn/by-example/tuples)
 - [Arrays](/learn/by-example/arrays)
 - [List sub typing](/learn/by-example/list-subtyping)
-
-[comment]: # (Add equality expression link)
```
Comment (rdulmina) [inline examples/list-equality/list_equality.md:14]: Why we just remove this comment instead of adding the relevant link

**Comment by rdulmina** [comment]
We need to address
https://github.com/rdulmina/ballerina-distribution/blob/0b2a4eda9c2043126184540d1e3cd6eaf705246d/examples/binary-data/binary_data.md?plain=1#L13

**Comment by SasinduDilshara** [comment]
> We need to address https://github.com/rdulmina/ballerina-distribution/blob/0b2a4eda9c2043126184540d1e3cd6eaf705246d/examples/binary-data/binary_data.md?plain=1#L13

We can address this comment after adding a BBE for `byte` type. 
I will be a separate task

**Review by sm1990 — DISMISSED** [review]

File: examples/http-service-file-upload/http_service_file_upload.md [hunk 2]
```diff
@@ -13,3 +13,4 @@ Run the service as follows.
 ## Related links
 - [`getByteStream()` - API documentation](https://lib.ballerina.io/ballerina/http/latest#Request#getByteStream)
 - [`http` module - Specification](/spec/http/#41-service-configuration)
+- [Binary Data](/learn/by-example/binary-data/)
```
Comment (MaryamZi) [inline examples/http-service-file-upload/http_service_file_upload.md:16]:
```suggestion
- [Binary data](/learn/by-example/binary-data/)
```

Comment (MaryamZi) [reply examples/http-service-file-upload/http_service_file_upload.md:16]: Let's update everywhere.

File: examples/yaml-to-anydata-with-projection/yaml_to_anydata_with_projection.md [hunk 3]
```diff
@@ -9,3 +9,6 @@ For more information on the underlying module, see the [`data.yaml` module](http
 ::: code yaml_to_anydata_with_projection.bal :::
 
 ::: out yaml_to_anydata_with_projection.out :::
+
+## Related links
+- [Binary Data](/learn/by-example/binary-data/)
```
Comment (MaryamZi) [inline examples/yaml-to-anydata-with-projection/yaml_to_anydata_with_projection.md:14]: We don't really have an example in this BBE, so not required?

**Review by MaryamZi — COMMENTED** [review]
The title mentions list equality, but it's not linked?

**Comment by MaryamZi** [comment]
> The title mentions list equality, but it's not linked?

Ping on this.

**Comment by SasinduDilshara** [comment]
> The title mentions list equality, but it's not linked?

will update

**Review by sm1990 — APPROVED** [review]

**Review by MaryamZi — DISMISSED** [review]

**Review by MaryamZi — COMMENTED** [review]
Why is the change https://github.com/ballerina-platform/ballerina-distribution/pull/5656/files#diff-08d375d409a03fb8f4e8e4393c3bcdff402cf831aff074d8a2fda0ebceacaa67R16 not here?

**Review by NipunaRanasinghe — COMMENTED** [review]
@SasinduDilshara FYA https://github.com/ballerina-platform/ballerina-distribution/pull/5656#pullrequestreview-2312299069

**Comment by SasinduDilshara** [comment]
> Why is the change https://github.com/ballerina-platform/ballerina-distribution/pull/5656/files#diff-08d375d409a03fb8f4e8e4393c3bcdff402cf831aff074d8a2fda0ebceacaa67R16 not here?

updated

**Comment by SasinduDilshara** [comment]
> Why is the change https://github.com/ballerina-platform/ballerina-distribution/pull/5656/files#diff-08d375d409a03fb8f4e8e4393c3bcdff402cf831aff074d8a2fda0ebceacaa67R16 not here?

updated

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**